### PR TITLE
Start of a FindOpenFOAM CMake module

### PR DESCRIPTION
**Pull request by pshriwise**

### CODE REVIEW

**Comment by pshriwise** [comment]
Made some small updates here that allow us to set properties of the `chtMultiRegionFoam` target instead of the global calls (`link_directories` and direct setting of the CMake CXX flags). By adding calls to find the full paths of the library via `find_library` we no longer need to set the link directories and can avoid this.

File: cmake/FindOpenFOAM.cmake [hunk 1]
```diff
@@ -0,0 +1,99 @@
+if(NOT DEFINED ENV{WM_PROJECT})
+  message(WARNING "Environment does not contain the necessary OpenFOAM variables.")
+  set(OpenFOAM_FOUND False)
+else()
+
+  set(OPENFOAM_LINK_DIRS $ENV{FOAM_LIBBIN})
+
+  message(STATUS "OpenFOAM ${OPENFOAM_LINK_DIRS}")
```
Comment (lebuller) [inline cmake/FindOpenFOAM.cmake]: We can probably remove this message statement on `OPENFOAM_LINK_DIRS` at this point

File: cmake/FindOpenFOAM.cmake [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  ${CHT_DIR}/fluid/compressibleCourantNo.C)
+
+  set(OPENFOAM_INCLUDE_DIRS
+  ${CHT_DIR}/
+  ${CHT_DIR}/fluid/
+  ${CHT_DIR}/solid/
+  ${CHT_DIR}/porousFluid/
+  ${CHT_DIR}/porousSolid/
+  ${CHT_DIR}/include/
+  $ENV{WM_PROJECT_DIR}/src/meshTools/lnInclude/
```
Comment (lebuller) [inline cmake/FindOpenFOAM.cmake]: Changing the `$ENV{WM_PROJECT_DIR}/src` to use `$ENV{FOAM_SRC}` instead may be a good plan. This goes for all of the libraries listed

File: cmake/FindOpenFOAM.cmake [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  add_executable(chtMultiRegionFoam ${CHT_DIR}/chtMultiRegionFoam.C ${OPENFOAM_SOURCES})
+  target_compile_definitions(chtMultiRegionFoam PUBLIC ${OPENFOAM_DEFINITIONS})
+  set_target_properties(chtMultiRegionFoam PROPERTIES LINK_FLAGS ${OPENFOAM_LINKER_FLAGS})
+  target_link_libraries(chtMultiRegionFoam PUBLIC ${OPENFOAM_LIBRARIES})
+  target_include_directories(chtMultiRegionFoam PUBLIC ${OPENFOAM_INCLUDE_DIRS})
+
+  ### FOR FUTURE USE, IMPORTED TARGET FOR OPENFOAM ###
+
+  # necessary for CMake < 3.12
+  # target_link_directories(chtMultiRegionFoam ${OPENFOAM_LINK_DIRS})
```
Comment (lebuller) [inline cmake/FindOpenFOAM.cmake:84]: Should we go ahead and make the same changes to the `openfoam-imported` target that were made to the `chtMultiRegionFoam` target in terms of how we are getting around the lack of `target_link_directories` in the current version of `cmake` on `ulam`

**Review by lebuller — CHANGES_REQUESTED** [review]

**Comment by pshriwise** [comment]
Just addressed a couple of these comments. With respect to the `openfoam-imported` target, as this is a working branch I think I'll leave it as is for now if that's ok with you. Once we're ready to actually use the target I'm thinking we can revisit it.

**Pull request merged**